src/components/header/Backtrace.js

```diff
@@ -0,0 +1,74 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// @flow
+
+import React, { PureComponent } from 'react';
+import { filterCallNodePathByImplementation } from '../../profile-logic/transforms';
+import { getOriginAnnotationForFunc } from '../../profile-logic/profile-data';
+
+import type { Thread, IndexIntoStackTable } from '../../types/profile';
+import type { CallNodePath } from '../../types/profile-derived';
+import type { CauseBacktrace } from '../../types/markers';
+import type { ImplementationFilter } from '../../types/actions';
+
+require('./Backtrace.css');
+
+type Props = {
+  thread: Thread,
+  cause: CauseBacktrace,
+  implementationFilter: ImplementationFilter,
+};
+
+type State = {};
+
+function callNodePathForStack(
+  thread: Thread,
+  stack: IndexIntoStackTable
+): CallNodePath {
+  const { stackTable, frameTable } = thread;
+  const path = [];
+  for (
+    let stackIndex = stack;
+    stackIndex !== null;
+    stackIndex = stackTable.prefix[stackIndex]
+  ) {
+    path.push(frameTable.func[stackTable.frame[stackIndex]]);
+  }
+  return path;
+}
+
+class Backtrace extends PureComponent<Props, State> {
```

Comment:
As said above, please leave out the State parameter.

Reply:
or instead maybe you can make this a functional component instead of a class component ?

Reply:
I've made it a functional component.